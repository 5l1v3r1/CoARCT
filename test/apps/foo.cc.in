/* Test file for function-mover.
 * To use, copy this to foo.cc, then run "../../build/apps/function-mover -p . foo.cc"
 * After, lines 7-10 should be removed (the closing bracket on 10 will be
 * removed, but not the newline.) See foo.cc.expected.
 *
 * To run properly, you will likely need to create a compile_database.json.
 * It will need to look like this:
[
{
  "directory": "/path/to/CoARCT/test/apps",
  "command": "/path/to/bin/clang++  -I/path/to/include/c++/v1 -I/path/to/lib/clang/5.0.1/include   -std=c++14   -fno-exceptions -fno-rtti -o CMakeFiles/corct.dir/foo.cc.o -c /path/to/CoARCT/test/apps/foo.cc",
  "file": "/path/to/CoARCT/test/apps/foo.cc"
},
]

 */
#include <iostream>

int foo(){
  std::cout << "foo!!\n";
  return 1;
}

double moo(){
  std::cout << "moo!!\n";
  return 2.0;
}
