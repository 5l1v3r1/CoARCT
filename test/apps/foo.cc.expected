/* Test file for function-mover.
 * To use, copy this to foo.cc, then run "../../build/apps/function-mover -p ."
 * After, lines 7-10 should be removed (the closing bracket on 10 will be
 * removed, but not the newline.)
 */
#include <iostream>



double moo(){
  std::cout << "moo!!\n";
  return 2.0;
}
